src/librustc_errors/emitter.rs

```diff
@@ -1226,6 +1247,30 @@ impl EmitterWriter {
             }
         }
     }
+
+    fn render_macro_backtrace_old_school(&self,
```

Comment:
`old_school` isn't very helpful here - what distinguishes it from a normal backtrace? `expanded` or `complete` or something might be better

Reply:
Yeah, I just used the name from the commit I'm partially reverting. I think it [looks](https://github.com/rust-lang/rust/pull/45545/files#diff-09f53becd1a865c0fd757a5eb39fd746) and bit "old school" (it doesn't have fancy span notes or anything).

Reply:
@durka would you be against using the full span machinery for them or would you think it'd be too verbose?

Reply:
The more spans the better, probably. I'm afraid I'm not too familiar with the new span hardware. What would it look like?

Reply:
Some changes will need to be made to the macro expansion machinery to avoid bogus spans (`<a macros>:1:36: 1:47 note: in this expansion of a! (defined in <a macros>)`), but I think it could be made to look something along the lines of:

```
error: expected one of `!`, `.`, `::`, `;`, `?`, `{`, `}`, or an operator, found `error`
  -->  a/src/lib.rs:6:16
3  |     () => { a!(@) };
   |             ----- in this expansion of `a!` (#2)
4  |     (@) => { a!(@@) };
   |              ------ in this expansion of `a!` (#3)
5  |     (@@) => {
6  |         syntax error;
   |                ^^^^^ expected one of 8 possible tokens here
  ::: src/main.rs
   |
12 |     a!();
   |     ----- in this expansion of `a!` (#1)

error: expected one of `!`, `.`, `::`, `;`, `?`, `{`, `}`, or an operator, found `error`
  --> src/main.rs:7:16
   |
4  |     () => { b!(@) };
   |             ----- in this expansion of `b!` (#2)
5  |     (@) => { b!(@@) };
   |              ------ in this expansion of `b!` (#3)
6  |     (@@) => {
7  |         syntax error;
   |                ^^^^^ expected one of 8 possible tokens here
13 |     b!();
   |     ----- in this expansion of `b!` (#1)

error: aborting due to 2 previous errors

error: Could not compile `b`.
```

Reply:
Well that mockup looks really good to me! :)

Reply:
@durka suggestion implemented in #46605. The output is not perfect as it is pointing at the pretty printed expanded macro code instead of the actual code, but it is close enough :)